plugins/inputs/chrony/chrony.go

```diff
@@ -388,22 +405,19 @@ func (c *Chrony) gatherSourceStats(acc telegraf.Accumulator) error {
 		}
 
 		// Trying to resolve the source name
-		sourceNameReq := fbchrony.NewNTPSourceNamePacket(sourceStats.IPAddr)
-		sourceNameRaw, err := c.client.Communicate(sourceNameReq)
-		if err != nil {
-			return fmt.Errorf("querying name of source %d failed: %w", idx, err)
-		}
-		sourceName, ok := sourceNameRaw.(*fbchrony.ReplyNTPSourceName)
-		if !ok {
-			return fmt.Errorf("got unexpected response type %T while waiting for source name", sourceNameRaw)
-		}
-
-		// Cut the string at null termination
 		var peer string
-		if termidx := bytes.Index(sourceName.Name[:], []byte{0}); termidx >= 0 {
-			peer = string(sourceName.Name[:termidx])
+
+		if sourceStats.IPAddr.String() == "::" {
+			// `::` IPs mean a local source directly connected to the server.
+			// For example a PPS source or a local GPS receiver.
+			// Then the name of the source is encoded in its RefID
+			// https://gitlab.com/chrony/chrony/-/blob/05bd4898a9a64fe1e5432a78939c4b4b82619c2f/client.c?page=3#L2152
```

Comment:
See above, remove the link to the source code and include the branching in `getSourceName`.